### DISCUSSION TITLE
C# Updating to Incremental Source Generators?

### DISCUSSION THREAD

**Discussion by hawkerm** [godotengine/godot-proposals#7106] [Scripting]
Didn't see any discussions on this so wanted to point out that `ISourceGenerator` is being deprecated, see https://github.com/dotnet/docs/issues/33757 - They can slowdown Visual Studio on large projects.

Incremental generators have replaced them: https://github.com/dotnet/roslyn/blob/main/docs/features/incremental-generators.md

I know the Godot source generators are effectively new, but wanted to point this out as it might become a bottleneck if larger C# projects get developed with Godot. There should probably be a plan in place to rewrite the current generators in the new API?

**Comment by paulloz** [comment]
There's been a draft about this for some time now (godotengine/godot#64899). Even if it isn't perfect, it moves us in the right direction.  

It'd probably be nice to have tests written for the generators, to be more confident future changes don't introduce regression.
